jbpm-flow/src/main/java/org/jbpm/workflow/instance/node/CompositeNodeInstance.java

```diff
@@ -429,6 +434,9 @@ public void setCurrentLevel(int currentLevel) {
     }
 
     protected boolean useAsync(final Node node) {
+        if (node.getMetaData().containsKey(NodeInstance.METADATA_WRAP_ASYNC_NODE)) {
+            return !((Boolean) node.getMetaData().get(NodeInstance.METADATA_WRAP_ASYNC_NODE));
+        }
```

Comment:
As I already pointed out - if this property is true - wrap async node - we are actually not wrapping anything.

Reply:
I understand the whole code except for this part. To me this will always return false in case of multi instance subprocess, which means that there will be no async node wrapped and subsequently no async node instance created (in the getNodeInstance() method of this class), so basically async mode won't be available for this type of subprocess node. Is that right or am I missing something? If it is not difficult to explain, why was this change needed?

Reply:
OK, probably got it. If it wasn't there, then if previous node was async, then it would wrap it as async event node and create again async event node instance (but we want the real node instance). Since the container is of type CompositeNodeInstance, we only need to handle it there. But still, afaik we use the from node instance to only compare its nodeId on lines 127-128 which should already point to the underlying actual node (since AsyncEventNode getId() delegates to the actual node and this is then preserved in async node instance as well).

Reply:
Not really. The problem was triggering
https://github.com/kiegroup/jbpm/pull/1465/files#diff-0d58c1b1aa778882879a180d7b1f90bdR402

now the trigger has the right node because of the behaviour in AsyncEventNodeInstance is different
https://github.com/kiegroup/jbpm/pull/1465/files#diff-1f9184540bbaf5380b28bb54bcb5fbaeR94

otherwise from node would be the same node triggering.

wrapping problem is a different problem.